vspace/src/main/java/edu/asu/diging/vspace/core/file/impl/StorageEngine.java

```diff
@@ -20,20 +29,23 @@
 @Component
 @PropertySource({"classpath:config.properties", "${appConfigFile:classpath:}/app.properties"})
 public class StorageEngine implements IStorageEngine {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
 
 	@Value("${uploads_path}")
 	private String path;
 	
 	/* (non-Javadoc)
-	 * @see edu.asu.diging.vspace.core.file.impl.IStorageEngine#storeFile(byte[], java.lang.String, java.lang.String)
-	 */
-	@Override
-	public String storeFile(byte[] fileContent, String filename, String directory) throws FileStorageException {
-		File parent = new File(path + File.separator + directory);
-		if (!parent.exists()) {
-			parent.mkdir();
-		}
-		File file = new File(parent.getAbsolutePath() + File.separator + filename);
+     * @see edu.asu.diging.vspace.core.file.impl.IStorageEngine#storeFile(byte[], java.lang.String, java.lang.String)
+     */
+    @Override
+    public String storeFile(byte[] fileContent, String filename, String directory, String path) throws FileStorageException {
```

Comment:
hm, no this shouldn't be changed. The point of the storage engine is to take the guess work out of where files are stored. Why did you add this? Is this because of multiple places that files need to be stored, or because there are a defined number of places files should be stored (e.g. images and exhibition exports)? I suspect it's the latter? In that case, there should be simply two instances of storage engine, setup with different base paths. The instances can be created in a config class and then injected with qualifiers.

Reply:
we can create 2 storage engines, one for uploads , one for downloads. 
Now in the downloads storage engine, the paths are dynamic. so the createFolder and storeFile methods will still need a path parameter. Or there will be methods such as createSpace, createModule, createSlide, etc